swap/migrations/20240615140942_btcpunished_update.sql

```diff
@@ -0,0 +1,128 @@
+UPDATE swap_states SET -- This query adds state3 to the BtcPunished state. (Introduced in PR #1676)
+    state = json_replace( -- Replaces "{"Alice":{"Done":"BtcPunished"}}" with "{"Alice": {"Done": "BtcPunished": {"state": <state3 object from BtcLocked>} }}"
+        state,
+        '$.Alice.Done',
+        json_object(
+            'BtcPunished',
+            (
+                SELECT json_extract(states.state, '$.Alice.BtcLocked') -- Read state3 object from BtcLocked
+                FROM swap_states AS states
+                WHERE
+                    states.swap_id = swap_states.swap_id -- swap_states.swap_id is id of the BtcPunished row
+                    AND json_extract(
+                        states.state, '$.Alice.BtcLocked'
+                    ) IS NOT NULL -- Filters out only the BtcLocked state. (json_extract returns null if property doesn't exist)
+            )
+        )
+    )
+WHERE json_extract(state, '$.Alice.Done') = 'BtcPunished'; -- Apply update only to BtcPunished state rows
+UPDATE swap_states SET -- This query adds state6 to the BtcPunished state. It then adds new attributes v (monero viewkey) and monero_wallet_restore_blockheight to state6. (Introduced in PR #1676)
+    state = json_replace(
+        state,
+        '$.Bob',  -- Replace '{"Bob":{"Done": {"BtcPunished": {"tx_lock_id":"..."} }}}' with {"Bob":{"BtcPunished":{"state":{<state6 object>}, "tx_lock_id": "..."}}}
+        json_object(
+            'BtcPunished', -- {"Bob":{"BtcPunished":{}}
+            json_object(
+                'state', -- {"Bob":{"BtcPunished":{"state": {}}}
+                json_insert(
+                    ( -- object that we insert properties into (original state6 from BtcCancelled state)
+                        SELECT json_extract(states.state, '$.Bob.BtcCancelled') -- Get state6 from BtcCancelled state.
+                        FROM swap_states AS states
+                        WHERE
+                            states.swap_id = swap_states.swap_id
+                            AND json_extract(
+                                states.state, '$.Bob.BtcCancelled'
```

Comment:
We can enter the `BtcPunished` state without ever entering the `BtcCancelled` state. Can you extract this data from somewhere else?